<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CondaEnvironmentConfigurationDialog</class>
 <widget class="QDialog" name="CondaEnvironmentConfigurationDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>548</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Conda Environment Configuration</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Environment Creator</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="condaLabel"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Logical Name:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="nameEdit">
       <property name="toolTip">
        <string>Enter a unique name for the virtual environment to register it with the Virtual Environment Manager</string>
       </property>
       <property name="placeholderText">
        <string>Name for registration of the virtual environment</string>
       </property>
       <property name="clearButtonEnabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>Target Environment Specification</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_11">
      <item row="0" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Name:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="condaNameEdit">
        <property name="toolTip">
         <string>Enter the name of the virtual environment in Conda</string>
        </property>
        <property name="placeholderText">
         <string>Name of the virtual environment in Conda</string>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Path:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="EricPathPicker" name="condaTargetDirectoryPicker" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="focusPolicy">
         <enum>Qt::FocusPolicy::WheelFocus</enum>
        </property>
        <property name="toolTip">
         <string>Enter the target directory for the conda environment</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>&lt;b&gt;Note:&lt;/b&gt; Only one of the above entries is mandatory.</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="condaSpecialsGroup">
     <property name="title">
      <string>Special Operations</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_8">
      <item row="0" column="0">
       <widget class="QRadioButton" name="condaCloneButton">
        <property name="toolTip">
         <string>Select to clone an environment</string>
        </property>
        <property name="text">
         <string>Clone Environment</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QRadioButton" name="condaRequirementsButton">
        <property name="statusTip">
         <string>Select to create the environment from a requirements file</string>
        </property>
        <property name="text">
         <string>from Requirements</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QFrame" name="condaCloneFrame">
        <property name="frameShape">
         <enum>QFrame::Shape::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Shadow::Raised</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_9">
         <item row="0" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Name:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="condaCloneNameEdit">
           <property name="toolTip">
            <string>Enter the name of the environment to be cloned</string>
           </property>
           <property name="clearButtonEnabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Path:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="EricPathPicker" name="condaCloneDirectoryPicker" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="focusPolicy">
            <enum>Qt::FocusPolicy::WheelFocus</enum>
           </property>
           <property name="toolTip">
            <string>Enter the directory of the environment to be cloned</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QFrame" name="condaRequirementsFrame">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="frameShape">
         <enum>QFrame::Shape::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Shadow::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="EricPathPicker" name="condaRequirementsFilePicker" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="focusPolicy">
            <enum>Qt::FocusPolicy::StrongFocus</enum>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Orientation::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>37</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_14">
       <property name="text">
        <string>Package Specs:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="condaPackagesEdit">
       <property name="toolTip">
        <string>Enter the package specifications for the environment separated by spaces</string>
       </property>
       <property name="clearButtonEnabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_10">
      <item row="0" column="0">
       <widget class="QLabel" name="versionLabel_3">
        <property name="text">
         <string>Python Version:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="condaPythonEdit">
        <property name="toolTip">
         <string>Enter the Python version for the environment</string>
        </property>
        <property name="clearButtonEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QCheckBox" name="condaInsecureCheckBox">
        <property name="text">
         <string>Allow insecure SSL connections</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QCheckBox" name="condaDryrunCheckBox">
        <property name="toolTip">
         <string>Select to perform just a dry-run</string>
        </property>
        <property name="text">
         <string>Perform dry-run</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::StandardButton::Cancel|QDialogButtonBox::StandardButton::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>EricPathPicker</class>
   <extends>QWidget</extends>
   <header>eric7/EricWidgets/EricPathPicker.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>nameEdit</tabstop>
  <tabstop>condaNameEdit</tabstop>
  <tabstop>condaTargetDirectoryPicker</tabstop>
  <tabstop>condaSpecialsGroup</tabstop>
  <tabstop>condaCloneButton</tabstop>
  <tabstop>condaRequirementsButton</tabstop>
  <tabstop>condaCloneNameEdit</tabstop>
  <tabstop>condaCloneDirectoryPicker</tabstop>
  <tabstop>condaRequirementsFilePicker</tabstop>
  <tabstop>condaPackagesEdit</tabstop>
  <tabstop>condaPythonEdit</tabstop>
  <tabstop>condaInsecureCheckBox</tabstop>
  <tabstop>condaDryrunCheckBox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>CondaEnvironmentConfigurationDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>644</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>CondaEnvironmentConfigurationDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>644</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
